File: arm.64/langs/fsharp/src/main.fs
namespace Pat

open System

module Program =
    [<EntryPoint>]
    let main args =
        let targetDir = if args.Length > 0 then Some args.[0] else None
        let dist = Booter.boot targetDir

        if not (List.isEmpty dist) then
            let trips = Tripper.trip ()
            for d in dist do
                for t in trips do
                    d.Pump.Invoke(t.Address, t.Payload, t.Options) |> ignore

        printfn "{\n  \"lang\": \"fsharp\",\n  \"status\": \"ready\",\n  \"engines\": %d\n}" dist.Length
        0

File: arm.64/langs/csharp/src/main.cs
namespace Pat;

public static class Program
{
    public static void Main(string[] args)
    {
        string? targetDir = args.Length > 0 ? args[0] : null;
        var dist = Booter.Boot(targetDir);

        if (dist.Count > 0)
        {
            var trips = Tripper.Trip();
            foreach (var d in dist)
            {
                if (d.Pump != null)
                {
                    foreach (var t in trips)
                    {
                        d.Pump(t.Address, t.Payload, t.Options);
                    }
                }
            }
        }

        Console.WriteLine($"{{\n  \"lang\": \"csharp\",\n  \"status\": \"ready\",\n  \"engines\": {dist.Count}\n}}");
    }
}
